coderd/userauth.go

```diff
@@ -191,12 +201,49 @@ func (api *API) userOAuth2Github(rw http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	_, created := api.createAPIKey(rw, r, database.InsertAPIKeyParams{
-		UserID:            user.ID,
-		LoginType:         database.LoginTypeGithub,
-		OAuthAccessToken:  state.Token.AccessToken,
-		OAuthRefreshToken: state.Token.RefreshToken,
-		OAuthExpiry:       state.Token.Expiry,
+	// This can happen if a user is a built-in user but is signing in
+	// with Github for the first time.
+	if link.UserID == uuid.Nil {
+		link, err = api.Database.InsertUserLink(ctx, database.InsertUserLinkParams{
+			UserID:            user.ID,
+			LoginType:         database.LoginTypeGithub,
+			LinkedID:          githubLinkedID(ghUser),
+			OAuthAccessToken:  state.Token.AccessToken,
+			OAuthRefreshToken: state.Token.RefreshToken,
+			OAuthExpiry:       state.Token.Expiry,
+		})
+		if err != nil {
+			httpapi.Write(rw, http.StatusInternalServerError, codersdk.Response{
+				Message: "A database error occurred.",
+				Detail:  xerrors.Errorf("insert user link: %w", err.Error).Error(),
```

Comment:
Honestly I'd just fmt.Sprintf here since you're just immediately converting it to a string.